File: spec/java/src/io/kaitai/struct/spec/TestFixedStruct.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.*;
import org.testng.annotations.Test;

import java.nio.charset.Charset;
import java.util.ArrayList;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertEqualsNoOrder;

public class TestFixedStruct extends CommonSpec {
    @Test
    public void testFixedStruct() throws Exception {
        FixedStruct r = FixedStruct.fromFile(SRC_DIR + "fixed_struct.bin");

        assertEquals(r.hdr().uint8(), 255);
        assertEquals(r.hdr().uint16(), 65535);
        assertEquals(r.hdr().uint32(), 4294967295L);
        //assertEquals(r.hdr().uint64(), 18446744073709551615);
        assertEquals(r.hdr().uint64(), 0xFFFFFFFFFFFFFFFFL);

        assertEquals(r.hdr().sint8(), -1);
        assertEquals(r.hdr().sint16(), -1);
        assertEquals(r.hdr().sint32(), -1);
        assertEquals(r.hdr().sint64(), -1);

        assertEquals(r.hdr().uint16le(), 66);
        assertEquals(r.hdr().uint32le(), 66);
        assertEquals(r.hdr().uint64le(), 66);

        assertEquals(r.hdr().sint16le(), -66);
        assertEquals(r.hdr().sint32le(), -66);
        assertEquals(r.hdr().sint64le(), -66);

        assertEquals(r.hdr().uint16be(), 66);
        assertEquals(r.hdr().uint32be(), 66);
        assertEquals(r.hdr().uint64be(), 66);

        assertEquals(r.hdr().sint16be(), -66);
        assertEquals(r.hdr().sint32be(), -66);
        assertEquals(r.hdr().sint64be(), -66);
    }

}

File: spec/cpp_stl/test_fixed_struct.cpp
#include <boost/test/unit_test.hpp>

#include <fixed_struct.h>

#include <iostream>
#include <fstream>

BOOST_AUTO_TEST_CASE(test_fixed_struct) {
//    FixedStruct* r = FixedStruct::fromFile("src/fixed_struct.bin");

    std::ifstream ifs("src/fixed_struct.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    fixed_struct_t* r = new fixed_struct_t(&ks);

    BOOST_CHECK_EQUAL(r->hdr()->uint8(), 255);
    BOOST_CHECK_EQUAL(r->hdr()->uint16(), 65535);
    BOOST_CHECK_EQUAL(r->hdr()->uint32(), 4294967295L);
    BOOST_CHECK_EQUAL(r->hdr()->uint64(), 18446744073709551615UL);

    BOOST_CHECK_EQUAL(r->hdr()->sint8(), -1);
    BOOST_CHECK_EQUAL(r->hdr()->sint16(), -1);
    BOOST_CHECK_EQUAL(r->hdr()->sint32(), -1);
    BOOST_CHECK_EQUAL(r->hdr()->sint64(), -1);

    BOOST_CHECK_EQUAL(r->hdr()->uint16le(), 66);
    BOOST_CHECK_EQUAL(r->hdr()->uint32le(), 66);
    BOOST_CHECK_EQUAL(r->hdr()->uint64le(), 66);

    BOOST_CHECK_EQUAL(r->hdr()->sint16le(), -66);
    BOOST_CHECK_EQUAL(r->hdr()->sint32le(), -66);
    BOOST_CHECK_EQUAL(r->hdr()->sint64le(), -66);

    BOOST_CHECK_EQUAL(r->hdr()->uint16be(), 66);
    BOOST_CHECK_EQUAL(r->hdr()->uint32be(), 66);
    BOOST_CHECK_EQUAL(r->hdr()->uint64be(), 66);

    BOOST_CHECK_EQUAL(r->hdr()->sint16be(), -66);
    BOOST_CHECK_EQUAL(r->hdr()->sint32be(), -66);
    BOOST_CHECK_EQUAL(r->hdr()->sint64be(), -66);

    delete r;
}
